search.go

```diff
@@ -402,6 +439,77 @@ func (l *Conn) Search(searchRequest *SearchRequest) (*SearchResult, error) {
 	}
 }
 
+// SearchAsync performs the given search request asynchronously, it returns a SearchAsyncResponse channel which will be
+// closed when the request finished and an error, not nil if the request to the server failed
+func (l *Conn) SearchAsync(ctx context.Context, searchRequest *SearchRequest) (<-chan *SearchAsyncResponse, error) {
+	msgCtx, err := l.doRequest(searchRequest)
+	if err != nil {
+		return nil, err
+	}
+	responses := make(chan *SearchAsyncResponse, 1)
```

Comment:
I dislike the idea to arbitrary limit or hardcode the channel here to a size of one. This would result in a blocking reader in case the function which called `SearchAsync` is processing the results or blocking due to other reasons. If we're working with a paging cookie, the LDAP DS could free the cookie to prevent Denial of Service and fetching further results would fail

Any thoughts on this, @go-ldap/committers?

Reply:
Directory servers have a short lease duration for paging cookies, @johnweldon. The overhead for the directory server is therefore very low, unless you intentionally start thousand paging requests without filters. It is more likely that the server will release the cookie. Queries with the PagingControl and the cookie will most likely fail.

Reply:
@Adphi - what is the benefit of a buffered channel here?

Reply:
I can't remember why, and I don't see any benefit of using a buffered channel here. I'll remove it.